### ISSUE TITLE
Identifying parent `cpgrid` for a `cpgrid_property`

### ISSUE THREAD

**Issue by HansKallekleiv** [equinor/fmu-dataio#278]
I am trying to create metadata for multiple grids / grid properties in roff format. (I.e. Drogon geogrid/simgrid)

Are there any tags that can be used to identify which grid a grid property belongs to? e.g. a `parent` keyword?

**Comment by perolavsvendsen** [comment]
`data.grid_model` is an existing tag, we could consider making this mandatory when content is `cpgrid_property`.

But I also think this more naturally belongs in the `fmu` section, similar to `realization` and `model` etc, something like:
```
data:
  content: cpgrid_property
fmu:
  grid_model: (mandatory if content is "cpgrid_property")
    name: "geogrid"
    (id: <in the future>)
    (uuid: <in the future>)
```

**Comment by perolavsvendsen** [comment]
Primary:
- Establish a sustainable link between `property` and `grid_model` so that consumers can link them
- ...with a minimum of config exposed to the end user.

**Comment by perolavsvendsen** [comment]
Offline discussion 2023-10-16
@daniel-sol @HansKallekleiv @perolavsvendsen 

Discussion isolated to "grid geometries and parameters produced by RMS".
Not discussed: Connecting grids across applications, multi-application grid results

Grids + parameters produced/used in FMU are represented by:
- Grid geometry
  - Represented by one (per grid) binary file on ROFF format.
- Grid parameters
  - Represented by multiple binary files on ROFF format.
  - Filename today: `<gridname>--<parameter_name>--<date>.roff`

**Proposal**

------------------------------------------------

***Export grid geometry:***

```python
mygridgeometry # <-- My grid geometry
exp = ExportData(
    name = "MyGridName"
    content = "grid_geometry"
exp.export(mygridgeometry)

```
--> Produces:
mygridname--geometry.roff
+ metadata:
```yaml
class: cpgrid
data:
  name: MyGridName
  content: grid_geometry
fmu:
  model:
  realization:
```

***Export parameters***

```python
mygridparameter # <-- My grid parameter, exported from the grid "MyGridName"
exp = ExportData(
    name: MyGridPropertyName
    content: {"grid_property": {"parent_name": "MyGridName}} # align with current convention
    )
exp.export(mygrid, parent="MyGridName") # the `parent` argument is currently supported, but only affects the filename.
```

- [ ] Parent currently supported, but only affects the filename (!). Need to include the parent input into metadata.

--> Produces:
`mygridpropertyname--mygridname.roff`  # NB should not break existing dependencies, may be backwards...
+ metadata:
```yaml
class: cpgrid
data:
  name: MyGridPropertyName
  content: grid_property
  grid_property:
    parent:
      name: MyGridName
fmu:
  model:
  realization:
```
------------------------------------------------

Convention is to have separate grid- and parameter files.

**Basic requirements (Webviz)**
- Grid parameter _values_ are linked to grid _geometry_ so that they can be co-visualized.
- An exported grid parameter can be linked to a corresponding grid geometry element

**Comment by HansKallekleiv** [comment]
For reference to the requirements above:
Below is an example of current RMS script for exporting grid geometry and parameters.
The `parent` argument is only used to create the filename. It is not used in the metadata.

```python
"""
Export geogrid and geogrid parameters (roff format)
Used for both visualization and as input to seismic project

Parameters required/used by seismic project is set in PROPS_SEISMIC
Additional paramters can be set with PROPS_OTHER
All parameters can be used for visualization (coviz)
"""
import xtgeo
import fmu.dataio as dataio
from pathlib import Path, PurePath
from fmu.config import utilities as utils

CFG = utils.yaml_load("../../fmuconfig/output/global_variables.yml")

PRJ = project

GNAME = "Geogrid"
PROPS_SEISMIC = ["PHIT", "VSH", "SW", "SG"]
PROPS_OTHER = ["KLOGH", "KV", "FACIES", "Region", "Zone", "Swl"]


def export_geogrid_parameters():
    """Export geogrid and associated parameters based on user defined lists """

    props = PROPS_SEISMIC + PROPS_OTHER

    edata = dataio.ExportData(
        config=CFG,
        content="depth",
        unit="m",
        vertical_domain={"depth": "msl"},
        workflow="rms grid and property model",
    )

    grd = xtgeo.grid_from_roxar(PRJ, GNAME)
    out = edata.export(grd, name=GNAME)
    print(f"Exported grid as {out}")

    # export properties
    for propname in props:
        prop = xtgeo.gridproperty_from_roxar(PRJ, GNAME, propname)
        out = edata.export(prop, name=propname, parent=GNAME)
        print(f"Exported grid property {propname:>13s}:    {out}")


if __name__ == "__main__":
    export_geogrid_parameters()
    print("Done.")

```
```bash
Exported grid as /project/fmu/tutorial/drogon/resmod/ff/users/hakal/webviz_case_1/share/results/grids/geogrid.roff
Exported grid property          PHIT:    /project/fmu/tutorial/drogon/resmod/ff/users/hakal/webviz_case_1/share/results/grids/geogrid--phit--geogrid.roff
Exported grid property           VSH:    /project/fmu/tutorial/drogon/resmod/ff/users/hakal/webviz_case_1/share/results/grids/geogrid--vsh--geogrid.roff
Exported grid property            SW:    /project/fmu/tutorial/drogon/resmod/ff/users/hakal/webviz_case_1/share/results/grids/geogrid--sw--geogrid.roff
Exported grid property            SG:    /project/fmu/tutorial/drogon/resmod/ff/users/hakal/webviz_case_1/share/results/grids/geogrid--sg--geogrid.roff
Exported grid property         KLOGH:    /project/fmu/tutorial/drogon/resmod/ff/users/hakal/webviz_case_1/share/results/grids/geogrid--klogh.roff
Exported grid property            KV:    /project/fmu/tutorial/drogon/resmod/ff/users/hakal/webviz_case_1/share/results/grids/geogrid--kv.roff
Exported grid property        FACIES:    /project/fmu/tutorial/drogon/resmod/ff/users/hakal/webviz_case_1/share/results/grids/geogrid--facies--geogrid.roff
Exported grid property        Region:    /project/fmu/tutorial/drogon/resmod/ff/users/hakal/webviz_case_1/share/results/grids/geogrid--region--geogrid.roff
Exported grid property          Zone:    /project/fmu/tutorial/drogon/resmod/ff/users/hakal/webviz_case_1/share/results/grids/geogrid--zone--geogrid.roff
Exported grid property           Swl:    /project/fmu/tutorial/drogon/resmod/ff/users/hakal/webviz_case_1/share/results/grids/geogrid--swl.roff
Done.
```

**Comment by HansKallekleiv** [comment]
```yaml
class: cpgrid_property
data:
  name: MyGridPropertyName
  content: inplace
  content_details:
     attribute: STOIIP
  grid_name: MyGridName
fmu:
  model:
  realization:
```

**Comment by perolavsvendsen** [comment]
Slightly zoomed out:

The general pattern in `fmu-dataio` today is as follows:

1) Make a Python object that represents the data you want to export
2) Initialize `fmu.dataio.ExportData` with some context
3) Give that Python object to `fmu.dataio.ExportData.export()`
4) Recognize and parse the given object, and extract more context directly
5) Generate metadata and save to disk

Given this pattern, it seems clear (?) that if we can create Python object representing the grid geometry + parameters, and make this a known object to `fmu-dataio`, we can infer the relationships between them.

e.g.

```python
from fmu.dataio import ExportData, RmsGrid  # <-- assuming part of fmu-dataio
mygrid = RmsGrid(PRJ.get_grid()) # pseudo-code
exp = ExportData()
exp.export(mygrid)

>>> Files on disk + metadata with context
```

**Comment by mferrera** [comment]
Can this be closed with #677 @jcrivenaes ?

**Comment by mferrera** [comment]
Answering my own question and considering this solved by ☝️
